### ISSUE TITLE
Fast Dial + TabMixPlus + BarTap, address bar no longer works

### ISSUE THREAD

**Issue by Teroan** [philikon/BarTab#25]
BarTap 1.3.2 update breaks address bar e.g. typing in or selecting url from history does not do anything. This only happens when Fast Dial (2.23b1) and TabMixPlus 0.3.8.2 are running at the same time with BarTap. Previous BarTap version had no problems.

**Comment by philikon** [comment]
Thanks for the feedback! However, I'm sorry, I can't reproduce this (Firefox 3.6, BarTab 1.3.x, Fast Dial 2.23b1, TMP 0.3.8.2). Perhaps you've configured Fast Dial or TMP differently than I have. I know that BarTab has some problems with TMP's own session restore mechanism. I really only want to support Firefox's mechanisms when there's feature duplication.

I'm going to close this for now. Feel free to provide me with more info, though. If I end up reproducing this bug, we can always reopen this issue.

**Comment by Teroan** [comment]
I'm running FF 3.5.8, which might be the cause of the problem. I tested again on a new profile and it still breaks down. Install order was BarTap, Fast Dial and then TabMixPlus using built-in session store, all where latest versions from FF own addon site.

I'll install the new FF 3.6 version and see if that helps. Didn't notice there even was new version out, so crossing fingers. :)

**Comment by Teroan** [comment]
FF 3.6 installed and still does not work. Not really sure whats going on here...

**Comment by philikon** [comment]
You can always access previous versions of BarTab using the "View All Versions" link on the bottom of https://addons.mozilla.org/en-US/firefox/addon/67651.

**Comment by Teroan** [comment]
Tested versions all the way to 1.2.2 which is the last version that seems to work for me. Again this was on a clean profile with FF 3.6, Fast Dial 2.23b1 and TabMixPlus 0.3.8.2 using FF own session restore.

**Comment by philikon** [comment]
I still can't reproduce this, but perhaps I'm doing it wrong. Here's what I did:
1. Start Firefox 3.6 with firefox -P, create a new profile
2. Install TMP 0.3.8.2 and BarTab 1.3.2.
3. Restart. Say "yes" when TMP asks whether to use Firefox's built-in session restore mechanism. Change no other settings (not in Firefox, not in BarTab, not in TMP)
4. Open a new tab, go to the address bar and start typing something. E.g. I typed "bartab" to find the addons.mozilla.org website in history. It found the entry and suggested in the drop down box. Used the down arrow to select the entry and hit enter. The page loaded.

Is that what you did? Perhaps I'm not quite understanding what you mean by "typing in or selecting url from history"...

**Comment by Teroan** [comment]
With "typing in or selecting url from history" i mean that i can type in the address but hitting enter will not do anything nor does selecting entries from the address bar drop down menu.

The problem happens ONLY when all of the three addons are installed, BarTap, TMP and Fast Dial. Removing ANY one of these and the problem goes away. Again, i'm not 100% sure the problem is BarTap or not, but after i updated BarTap i got this problem.

Update:
Just tested on another machine running same setting as my own and i get the same problem. As soon as the three addons are installed, address bar no longer works.

Not sure if it matters but i'm running FF in Windows XP Pro SP3 with latest updates.

**Comment by Teroan** [comment]
After FF update to 3.6.2, BarTap 1.2.2 no longer works. :(

I get error below in "Error Console" when trying to enter url or pick one from the drop menu. I'm using FF 3.6.2 with Fast Dial 2.23b1 , BarTap 1.4.2 and TMP 3.8.2 ( + some dev addons).

Error: errorHook: this.TMPisBlankBrowser is not a function@chrome://bartap/content/browser.js.62

Source file: file:///C:/Documents%20and%20Settings/Teroan/Application%20Data/Mozilla/Firefox/Profiles/k2mdubb3.Developer/extensions/chromebug@johnjbarton.com/components/chromebug-startup-observer.js
Line: 240

---

Error: errorHook: TypeError: this.TMPisBlankBrowser is not a function@chrome://bartap/content/browser.js.62

Source file: file:///C:/Documents%20and%20Settings/Teroan/Application%20Data/Mozilla/Firefox/Profiles/k2mdubb3.Developer/extensions/chromebug@johnjbarton.com/components/chromebug-startup-observer.js
Line: 240

**Comment by Teroan** [comment]
I found the problem. It appears that Fast Dial uses the "isBlankBrowser" which you have already dealth with TMP case. Specifically the FD uses this is for "show fast dial on blank tabs" which i don't use so i simply comment out the isBlankBrowser code from "fdloader.js" that is part of the Fast Dial extension. I'm no programmer but what i can tell everything works fine now.

**Comment by philikon** [comment]
pug pug reports similar things:

"I found a bug if BarTab (>1.3.0) is used together with FastDial and TabMixPlus Extensions.
Last working version of BarTab is 1.2.2

Enviroment: Fresh instal of Firefox 3.6.2 with FastDial 2.23b1, TabMixPlus 0.3.8.2 and BarTab 1.4bb6 with default settings

Symptoms: FF is not loading entered url if selected tab is already loaded, it is functioning properly only for newly opened tabs

Probably Fix to problem is: (changes are marked with "===>")

```
/* Tab Mix Plus compatibility: It likes reusing blank tabs.  In doing
   so it confuses tabs on the bar tab with blank ones.  Fix that. */
if (tabbrowser.isBlankBrowser) {
```

===>      BarTap.isBlankBrowser = tabbrowser.isBlankBrowser;
      tabbrowser.isBlankBrowser = function (aBrowser) {
        if (aBrowser.getAttribute("ontap") == "true") {
          return false;
        }
===>        return BarTap.isBlankBrowser(aBrowser);
      };
    }
"

**Comment by Teroan** [comment]
That worked nicely, thanks!

**Comment by philikon** [comment]
Improve compatibility with the Fast Dial + Tab Mix Plus combination of add-ons.

Closed by 041c937a8308b3e26769a9faadd1b0cd4a097be6
